### PR TITLE
Submit  TDD_Design  "estimation-of-COVID-19-pandemic.ipynb"

### PR DESCRIPTION
**Pull request by WHQWHQWHQ**
# Description

Format all codes, add a new test case

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [x] review the artifacts from local build
- [x] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [x] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have applied tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by zhangqi444** [comment]
Please give a better title to the PR always.

File: open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "import matplotlib.pyplot as plt\n",
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>Remove line 20 (two empty lines, should be one)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>remove | at the end of the line</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #1.    def get_onecolumn_groupby_sum(df, column_name):
</pre><p>name this function as <code>def groupby_sum()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #6.                'c1': [1, 2, 3, 4, 5], 
</pre><p> </p><p>c1': [1, 1, 1, 2, 2],&nbsp;</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #11.    class TestOnecolumnGroupbySum(unittest.TestCase):
</pre><p>TestYourFunctionName. What's OnecolumnGroupbySum?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #22.            assert actual_result.iloc[0, 0] == 6
</pre><p>why not compare the entire actual_result?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>and aggregation function <code>sum()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #1.    def plot_infected_vs_recovered():
</pre><p>make the US as a parameter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #15.        def test_drop_columns_happy_case(self):
</pre><p>give an empty line before this line.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #23.            assert test_df.iloc[0, 0] == 11
</pre><p>You should check if the result is </p><pre class="ql-syntax" spellcheck="false">d.DataFrame(
&nbsp; &nbsp; &nbsp; &nbsp; {
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "c3": [11, 12, 13, 14, 15],
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; "c4": [16, 17, 18, 19, 20],
&nbsp; &nbsp; &nbsp; &nbsp; }
&nbsp; &nbsp; )


</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #3.            raise Exception("df cannot be None.")
</pre><p>infected df cannot be None. Please make the error message to be more specific.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #28.    class TestDropColumns(unittest.TestCase):
</pre><p>are you testing drop_columns here? I think you are testing <code>mkframe()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #39.            assert test_df.iloc[1, 1] == 9
</pre><p>You should compare the entire pandas series, instead of a single value. Please apply this to all test cases.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>filter_by_year_and_month(). Let's always try not to put the plot code into the function if possible.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #1.    def filter_by_country_region(df, country_region):
</pre><p>countries_and_region</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #1.    def get_rt_ax(df):
</pre><p>def rt_with_na_filled()</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>Just create a function called <code>get_()</code> for all the code here before line 12</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <pre class="ql-syntax" spellcheck="false">Line #5.    def diff(df):
</pre><p>You have 3 functions to get diff. This one and  <code>plot_df_column_diff</code> and  <code>plot_column_to_show_by_getting_column_to_diff</code>. Could you please merge them?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>get_smoothed_ax()</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>YES</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>Finished</p><p> </p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>done</p><p> </p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>	DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DOE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>Sorry,DONE IT</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>this issue is still there. Please fix.</p>

Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>Fix the test class name</p>

File: open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb [hunk 2]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <p>what's this change for?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <p>Please clear all the output of this notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #1.    %%ipytest -qq
</pre><p>Please always format the code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #8.        def test_filter_with_happy_case(self):
</pre><p>You should not say test with happy case. You can only say test with a certain input, e.g. an invalid input. Remove the unncessary with from the test case name. Please apply this to all the others.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #20.        def test_filter_with_invalid_number(self):
</pre><p>can you add some other test cases? E.g. with invalid column name, with empty df, etc.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #13.        def test_groupby_sum_happy_case(self):
</pre><p>This is a good name!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #34.            #act
</pre><p><code> # act &amp; assert</code>  Please apply to all the other similar cases.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #23.            test_dropped_df=pd.DataFrame(
</pre><p>expected_result = </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #33.            from pandas.testing import assert_frame_equal
</pre><p>always have import at the beginning of the cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #34.            assert_frame_equal(test_df,test_dropped_df)
</pre><p>Please format all your cells.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #49.                get_drop_columns(None, test_df_2, test_df_3, "US")
</pre><p>what's this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <pre class="ql-syntax" spellcheck="false">Line #69.                get_onecolumn_groupby_sum(test_df_1, test_df_2, test_df_3, "China")
</pre><p>what's this? Please carefully review all your code first before submit a PR.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

File: open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb (hunk 1 above)
Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>FINISHED</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:25]: <p>DONE</p>

File: open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb (hunk 2 above)
Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <p>a wrong....</p><p> </p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <p>Done</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:4]: <p>Done</p>

Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb]: <p>...  <code>test_filter_with_a_certain_input</code> -&gt; <code>test_filter_happy_case</code></p>

File: open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb [hunk 3]
```diff
@@ -23,9 +23,11 @@
     "import numpy as np\n",
```
[the remark this answers is not in the document]
Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <p>DONE</p>

[the remark this answers is not in the document]
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: If you didt address this, please not resolve the comment.

[the remark this answers is not in the document]
Comment (zhangqi444) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <p>You have 3 input parameters here. You need to have test cases for all 3 right? Please test None, invalid, empty for all three parameters if possible.</p>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <pre class="ql-syntax" spellcheck="false">Line #1.    def column_to_show_by_getting_column_to_diff(
</pre><p>change the function to <code>def append_diff_column(df, new_column, column_to_diff)</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <pre class="ql-syntax" spellcheck="false">Line #50.        def test_column_to_show_by_getting_column_to_diff_with_none_column_to_diff(
</pre><p>Did you run all your code? Are you sure this test case will work? You are not testing column_to_diff is None. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/170/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

**Comment by zhangqi444** [comment]
Could you please help resolve the exact comments you have addressed?

File: open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb (hunk 3 above)
Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <p>DONE</p>

[the remark this answers is not in the document]
Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <p>DONE</p>

Comment (WHQWHQWHQ) [reply open-machine-learning-jupyter-book/assignments/data-science/estimation-of-COVID-19-pandemic.ipynb:23]: <p>Sorry, Confused, Done</p>

**Review by zhangqi444 — APPROVED** [review]
